erpnext/accounts/report/summarized_profit_and_loss/summarized_profit_and_loss.py

```diff
@@ -246,66 +247,142 @@ def calculate_group_totals(self, group, company, report_date, month_start, year_
 
 	def get_account_balances(self, account, company, report_date, month_start, year_start,
 			prev_year_date, prev_year_month_start, prev_year_start):
+		"""Get account balances with sign based on Account Group's root type."""
 		account_doc = frappe.get_doc("Account", account)
-		fiscal_year = report_date.year
+		
+		# Get Account Group's root type
+		group_root_type = frappe.get_value(
+			"Account Group",
+			frappe.get_value("Account Group Row", {"account": account}, "parent"),
+			"root_type"
+		)
+
+		# Get all balances
+		balances = {
+			"mtd_actual": self.get_balance(account, company, month_start, report_date, account_doc),
+			"ytd_actual": self.get_balance(account, company, year_start, report_date, account_doc),
+			"mtd_prev_year": self.get_balance(account, company, prev_year_month_start, prev_year_date, account_doc),
+			"ytd_prev_year": self.get_balance(account, company, prev_year_start, prev_year_date, account_doc),
+			"mtd_budget": self.get_budget_amount(account, company, month_start, report_date, year_start, report_date.year),
+			"ytd_budget": self.get_budget_amount(account, company, year_start, report_date, year_start, report_date.year)
+		}
+
+		# Apply sign based on group type (positive for Income, negative for Expense)
+		multiplier = 1 if group_root_type == "Income" else -1
+		signed_balances = {k: abs(v) * multiplier for k, v in balances.items()}
 
 		return {
 			"row_type": "Account",
 			"account_name": f"{account_doc.account_number} - {account_doc.account_name}" if account_doc.account_number else account_doc.account_name,
 			"account": account,
 			"account_type": account_doc.account_type,
-			"mtd_actual": self.get_balance(account, company, month_start, report_date, account_doc),
-			"mtd_budget": self.get_budget_amount(account, company, month_start, report_date, year_start, fiscal_year),
-			"mtd_prev_year": self.get_balance(account, company, prev_year_month_start, prev_year_date, account_doc),
-			"ytd_actual": self.get_balance(account, company, year_start, report_date, account_doc),
-			"ytd_budget": self.get_budget_amount(account, company, year_start, report_date, year_start, fiscal_year),
-			"ytd_prev_year": self.get_balance(account, company, prev_year_start, prev_year_date, account_doc)
+			**signed_balances
 		}
 
+	def get_tree_descendants(self, doctype, parent_names):
+		"""Get descendants for tree DocType nodes"""
+		if not parent_names:
+			return []
+		
+		parent_data = frappe.get_all(doctype, 
+			filters={'name': ['in', parent_names]},
+			fields=['lft', 'rgt'],
+			order_by='lft'
+		)
+		if not parent_data:
+			return []
+
+		or_filters = []
+		args = []
+		for p in parent_data:
+			or_filters.append("(lft >= %s AND rgt <= %s)")
+			args.extend([p.lft, p.rgt])
+
+		return [d.name for d in frappe.db.sql(
+			f"""SELECT name FROM `tab{doctype}` WHERE {" OR ".join(or_filters)}""",
+			args, as_dict=1
+		)]
```

Comment:
we already have many utilities for this, including `get_dimension_with_children`